src/rust/ide/lib/view/src/graph_editor/component/project_name.rs

```diff
@@ -0,0 +1,327 @@
+// === Frp ===
+// ===========
+
+#[derive(Debug,Clone,CloneRef)]
+#[allow(missing_docs)]
+pub struct Frp {
+    pub inputs  : FrpInputs,
+    pub outputs : FrpOutputs,
+    pub network : frp::Network,
+}
+
+impl Deref for Frp {
+    type Target = FrpInputs;
+    fn deref(&self) -> &Self::Target {
+        &self.inputs
+    }
+}
+
+impl Default for Frp {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Frp {
+    /// Create new Frp.
+    pub fn new() -> Self {
+        let network = frp::Network::new();
+        let inputs  = FrpInputs::new(&network);
+        let outputs = FrpOutputs::new(&network);
+        Self{network,inputs,outputs}
+    }
+}
+
+
+
+// ==================
+// === Animations ===
+// ==================
+
+/// ProjectName's animations handlers.
+#[derive(Debug,Clone,CloneRef)]
+pub struct Animations {
+    opacity  : Animation<f32>,
+    position : Animation<Vector3<f32>>
+}
+
+impl Animations {
+    /// Create new animations handlers.
+    pub fn new(network:&frp::Network) -> Self {
+        let opacity = Animation::new(&network);
+        let position = Animation::new(&network);
+        Self{opacity,position}
+    }
+}
+
+
+
+// ========================
+// === ProjectNameModel ===
+// ========================
+
+#[derive(Debug,Clone,CloneRef)]
+#[allow(missing_docs)]
+pub struct ProjectNameModel {
+    logger         : Logger,
+    animations     : Animations,
+    display_object : display::object::Instance,
+    view           : component::ShapeView<background::Shape>,
+    text_field     : TextField,
+    project_name   : Rc<RefCell<String>>,
+    name_output    : frp::Source<String>,
+}
+
+impl ProjectNameModel {
+    /// Create new ProjectNameModel.
+    pub fn new(world:&World,frp:&Frp) -> Self {
+        let scene                 = world.scene();
+        let logger                = Logger::new("ProjectName");
+        let display_object        = display::object::Instance::new(&logger);
+        let font                  = scene.fonts.get_or_load_embedded_font("DejaVuSansMono").unwrap();
+        let size                  = Vector2::new(scene.camera().screen().width,TEXT_SIZE);
+        let base_color            = TRANSPARENT_TEXT_COLOR;
+        let text_size             = TEXT_SIZE;
+        let text_field_properties = TextFieldProperties{base_color,font,size,text_size};
+        let text_field            = TextField::new(&world,text_field_properties);
+        let view_logger           = Logger::sub(&logger,"view_logger");
+        let view                  = component::ShapeView::<background::Shape>::new(&view_logger, scene);
+        let project_name          = Rc::new(RefCell::new(UNKNOWN_PROJECT_NAME.to_string()));
+        let renamed_output        = frp.outputs.name.clone();
+        let animations            = Animations::new(&frp.network);
+        Self{logger,view,display_object,text_field,project_name,
+            name_output: renamed_output
+            ,animations}.init()
+    }
```

Comment:
this is soooo ugly. Please, rename the var instead.

Reply:
omg idk what happened here